app/controllers/concerns/api/compatibility_checker.rb

```diff
@@ -4,9 +4,11 @@ module CompatibilityChecker
 
     # removes unsupported "nested" flag from "host_parameters_attributes" (both Array and Hash formats supported)
     def check_create_host_nested
-      params[:host] && params[:host][:host_parameters_attributes] && params[:host][:host_parameters_attributes].each do |attribute|
-        attribute = attribute[1] if attribute.is_a? Array
-        Foreman::Deprecation.api_deprecation_warning("Field host_parameters_attributes.nested ignored") unless attribute.delete(:nested).nil?
+      if params[:host] && params[:host][:host_parameters_attributes] && (attrs = params[:host][:host_parameters_attributes])
```

Comment:
Why not `if params[:host] && (attrs = params[:host][:host_parameters_attributes])` ?